exp/addons/api/v1alpha3/clusterresourcesetbinding_types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ANCHOR: ResourceBinding
+
+// ResourceBinding keeps status of a resource that is applied to the cluster that ClusterResourceSetBinding belongs to.
+type ResourceBinding struct {
+	// Hash is the hash of a resource's data. This can be used to decide if a resource is changed.
+	// For "ApplyOnce" ClusterResourceSet.spec.mode, this is no-op as that mode does not act on change.
+	Hash string `json:"hash,omitempty"`
+	// LastAppliedTime identifies when this resource was last applied to the cluster.
+	// +optional
+	LastAppliedTime *metav1.Time `json:"lastAppliedTime,omitempty"`
+	// Applied is to track if a resource is applied to the cluster or not.
+	Applied bool `json:"applied,omitempty"`
+}
+
+// ANCHOR_END: ResourceBinding
+
+// ResourcesSetBinding keeps info on all of the resources in a ClusterResourceSet.
+type ResourcesSetBinding struct {
+	// Resources is a map of Secrets/ConfigMaps and their ResourceBinding.
+	// The map's key is a concatenated string of form: <resource-type>/<resource-name>.
+	Resources map[string]ResourceBinding `json:"resources,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=clusterresourcesetbindings,scope=Namespaced,categories=cluster-api
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+
+// ClusterResourceSetBinding lists all matching ClusterResourceSets with the cluster it belongs to.
+type ClusterResourceSetBinding struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// ClusterResourceSetMap is a map of ClusterResourceSet and its resources which is also a map.
+	ClusterResourceSetMap map[string]ResourcesSetBinding `json:"clusterresourcesetmap,omitempty"`
```

Comment:
Is the key of this map the Cluster name? The comments on the field and the struct are a little confusing